### ISSUE TITLE
exit code 1 - already up-to-date

### ISSUE THREAD

**Issue by abix5** [mooltiverse/nyx#194]
```
time="2023-04-17T15:01:50Z" level=debug msg="command 'MARK' is not up to date, running..."
time="2023-04-17T15:01:50Z" level=debug msg="running the Mark command..."
time="2023-04-17T15:01:50Z" level=debug msg="the release type has the git commit flag disabled"
time="2023-04-17T15:01:50Z" level=debug msg="the release type has the git tag flag enabled"
time="2023-04-17T15:01:50Z" level=debug msg="repository latest commit in HEAD branch is '142196c2588e90baa2508e330e4896d6b856d352'"
time="2023-04-17T15:01:50Z" level=debug msg="tagging latest commit '142196c2588e90baa2508e330e4896d6b856d352' with tag 'v0.8.0'"
time="2023-04-17T15:01:50Z" level=debug msg="tagging as 'v0.8.0'"
time="2023-04-17T15:01:50Z" level=debug msg="repository latest commit in HEAD branch is '142196c2588e90baa2508e330e4896d6b856d352'"
time="2023-04-17T15:01:50Z" level=debug msg="tag 'v0.8.0' applied to commit '142196c2588e90baa2508e330e4896d6b856d352'"
time="2023-04-17T15:01:50Z" level=debug msg="the release type has the git push flag enabled"
time="2023-04-17T15:01:50Z" level=debug msg="pushing local changes to remotes"
time="2023-04-17T15:01:50Z" level=debug msg="pushing local changes to remote 'origin'"
time="2023-04-17T15:01:50Z" level=debug msg="looking up credentials for remote 'origin'"
time="2023-04-17T15:01:50Z" level=debug msg="using configured credentials for remote 'origin'"
time="2023-04-17T15:01:50Z" level=debug msg="attempting push to 'origin' using user name and password credentials."
time="2023-04-17T15:01:50Z" level=debug msg="pushing changes to remote repository 'origin' using username and password"
time="2023-04-17T15:01:50Z" level=debug msg="username and password authentication will use custom authentication options"
an error occurred when trying to push: already up-to-date
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: command terminated with exit code 1
```

If you run the pipeline again, and try to release the tag, the script crashes with an error. The logs show that nyx says that everything is updated. The exit code in this case should be `0`

**Comment by flelli** [comment]
Thanks @abix5 , can you please send the log of the second run?

**Comment by abix5** [comment]
This is the log of the second start, the first one passes with success.
@flelli

**Comment by flelli** [comment]
@abix5 , any chance you forgot to attach the log?

**Comment by flelli** [comment]
Closing this issue due to missing information. I'll be glad to reopen upon feedback.

**Comment by abix5** [comment]
I attached the log in the first message. It is the same except that during the first run everything goes correctly, the version number tag is pushed. But the second time, nyx sees the tag on this commit and says "already up-to-date" and exits with code 1! I attached the log of the second run specifically. I have nothing else to add here!


----
Again, I have a very big objection that there was actually no error, but nyx exits with a code of 1. In my opinion, this is incorrect behavior, because regardless of the settings and regardless of which branch I run it on, if it did not encounter any errors, the exit code should be 0.
`
$ nyx infer
time="2023-05-04T14:46:16Z" level=debug msg="selecting the command to run"
time="2023-05-04T14:46:16Z" level=debug msg="command to run is INFER"
time="2023-05-04T14:46:16Z" level=debug msg="Nyx.run(INFER)"
time="2023-05-04T14:46:16Z" level=debug msg="Nyx.infer()"
time="2023-05-04T14:46:16Z" level=debug msg="running command 'INFER'"
time="2023-05-04T14:46:16Z" level=debug msg="looking up command instance 'INFER' from cache"
time="2023-05-04T14:46:16Z" level=debug msg="no command instance 'INFER' found in cache"
time="2023-05-04T14:46:16Z" level=debug msg="creating new command instance 'INFER'"
time="2023-05-04T14:46:16Z" level=debug msg="instantiating the Git repository in '/builds/front/crm-frontend'"
time="2023-05-04T14:46:16Z" level=warning msg="the resume flag has been set and the state file has been configured but no state file exists at the given location /builds/front/crm-frontend/.nyx-state.yml. The state file will not be resumed."
time="2023-05-04T14:[46](https://gitlab.****.ru/front/crm-frontend/-/jobs/12721#L46):16Z" level=debug msg="instantiating the initial state"
time="2023-05-04T14:46:16Z" level=debug msg="new Infer command object"
time="2023-05-04T14:46:16Z" level=debug msg="checking whether the Infer command is up to date"
time="2023-05-04T14:46:16Z" level=debug msg="the Infer command is not up to date because the internal state has no version yet or the state version doesn't match the version previously generated by Infer"
time="2023-05-04T14:46:16Z" level=debug msg="command 'INFER' is not up to date, running..."
time="2023-05-04T14:46:16Z" level=debug msg="running the Infer command..."
time="2023-05-04T14:46:16Z" level=debug msg="clearing the state from Infer outputs"
time="2023-05-04T14:46:16Z" level=debug msg="resolving the release type among enabled ones"
time="2023-05-04T14:46:16Z" level=debug msg="evaluating release type: 'mainline'"
time="2023-05-04T14:46:16Z" level=debug msg="release type 'mainline' specifies a match branches template '^(master|main)$' that evaluates to regular expression: '^(master|main)$'"
time="2023-05-04T14:46:16Z" level=debug msg="current branch 'feat-4' not matched by release type 'mainline' matchBranches regular expression '^(master|main)$'. Skipping release type 'mainline'"
no suitable release types have been configured or none of the configured release types matches the current environment
Uploading artifacts for failed job
00:02
Uploading artifacts...
WARNING: build.env: no matching files. Ensure that the artifact path is relative to the working directory (/builds/front/crm-frontend) 
ERROR: No files to upload                          
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: command terminated with exit code 1
`
